static/test/javascripts/spec/common/onward/breaking-news.spec.js

```diff
@@ -0,0 +1,286 @@
+/*eslint no-console: 0*/
+define([
+    'common/utils/config',
+    'helpers/injector',
+    'common/utils/$',
+    'common/utils/storage',
+    'common/utils/ajax',
+    'lodash/objects/defaults',
+    'Promise'
+], function (
+    config,
+    Injector,
+    $,
+    storage,
+    ajax,
+    defaults,
+    Promise
+) {
+    describe('Breaking news', function () {
+        var injector = new Injector(),
+            breakingNewsURL = (config.switches.breakingNewsFromAdminJobs) ? '/news-alert/alerts' : '/breaking-news/lite.json',
+            knownAlertIDsStorageKey = 'gu.breaking-news.hidden',
+            server;
+
+        function alertThatIs(type, options) {
+            options = defaults(options || {}, {
+                collection: 'uk',
+                age: 1
+            });
+            var sAge = options.age !== 1 ? options.age + 'min ' : '';
+            return {
+                href: options.collection,
+                content: [{
+                    headline: sAge + options.collection + ' ' + type + ' headline',
+                    trailText: options.collection + ' ' + type + ' trailText',
+                    id: options.collection + '_' + type,
+                    frontPublicationDate: Date.now() - (1000 * 60 * options.age)
+                }]
+            };
+        }
+
+        function mockBreakingNewsWith(collections) {
+            return new Promise(function (resolve, reject) {
+                injector.mock({
+                    'common/utils/storage': storage,
+                    'common/utils/ajax': ajax
+                }).require(['common/modules/onward/breaking-news'], function (breakingNews) {
+                    server = sinon.fakeServer.create();
+                    server.respondImmediately = true;
+                    server.autoRespond = true;
+                    server.respondWith(breakingNewsURL, [200, {'Content-Type': 'application/json'}, JSON.stringify({
+                        webTitle: 'Breaking News',
+                        collections: collections
+                    })]);
+                    breakingNews().then(function(result) {
+                        // make sure the DOM has finished updating
+                        requestAnimationFrame(function () {
+                            resolve(result);
+                        });
+                    }, reject);
+                    server.restore();
+                }, function (e) {
+                    console.log(e);
```

Comment:
is it not? :)

it's just because it turns out we override `inject.require` with a promised version, and for ages i couldn't work out why nothing was working. turns out the promise i didn't know existed was rejecting but i wasn't catching it i.e. it was test code issue, not code that was being tested. just there incase it happens again.